terms/evaluationmethods_ae.tex

```diff
@@ -0,0 +1,131 @@
+\chapter{Evaluation Methods}
+\label{ch:evaluation-methods}
+\index{evaluation methods|(}
+
+Although modern machine learning is exceptionally diverse and multidisciplinary, what unifies the field is standard experimental evaluation methodologies that are adopted based on the task at hand~\citep{Kibler1988}.
+%
+Evaluation is involved both in model selection, i.e.\ during hyperparameter tuning, and in comparing multiple models against each other or a random baseline.
+Ideally, evaluation is aimed at interpreting a model's behaviour and analysing its errors~\citep{Doshi2017}.
+%
+\citet{Japkowic2011} outline four essentials of evaluation methodology: a test set, performance measures, error estimation, and statistical significance tests.
+
+When evaluating machine learning models, it is common to rely on some annotated \emph{gold standard}\index{evaluation methods!gold standard} dataset that serves as a proxy of real-world phenomena~\citep{Manning2008}.
+%
+Examples of such datasets can be found in centralised collections such as the UCI Machine Learning repository~\citep{Dua2017} or online communities like Kaggle\footnote{\url{https://www.kaggle.com}}.
+However, one needs to be aware that a dataset
+may not represent the whole domain~\citep{Japkowic2011},
+contain annotation artefacts,
+or might have been constructed with vague class definitions~\citep{Hand2006}.
+
+While there is an assumption that training and testing sets come from the same distribution in order for a model to work as expected~\citep{Japkowic2011}, it is essential that the model is tested on previously unseen data.
+Otherwise, it yields overoptimistic results~\citep{Kibler1988}.
+%
+Error estimation techniques such as a held-out\index{cross-validation!holdout} set or cross-validation\index{cross-validation} are detailed
+% \nameref{ch:cross-validation},
+on page~\pageref{ch:cross-validation}, so the following is focussed on evaluation metrics and statistical tests.
+
+\section{Performance Measures}
+\citet{Stapor17} notes that according to the \emph{no free lunch theorem}\index{no free lunch theorem}~\citep{Wolpert1996}, it is unreasonable to expect that one classifier is generally better than another since there will always be problems where one outperforms the other and vice versa.
+Evaluation is also highly task-specific and selecting a performance measure should be done with the end goal in mind~\citep{goodfellow2016deeplearning}. For example, natural language generation models used in image captioning are penalised for ``hallucinating'' extra information, i.e.\ such models are considered worse, but this aspect can be encouraged when applied to poetry generation.
+
+\subsection{Automatic Evaluation Metrics}
+For supervised classification, the simplest measure is \emph{accuracy}\index{confusion matrix!accuracy} which is the ratio of correctly predicted examples, and its reverse metric \emph{error rate}\index{evaluation methods!error rate}, i.e.\ the fraction of errors.
```

Comment:
This index entry should not be under confusion matrix -- but under evaluation methods !!!  Please fix.

Reply:
Done. I wanted to reuse the previously added index for accuracy, that's why it was under confusion matrix.